### PR TITLE
[Merged by Bors] - feat: add Display for ClusterCondition

### PR DESCRIPTION
**Pull request by Techassi**
This PR adds a `Display` impl for `ClusterCondition`. It might be worth a thought that we want to provide different `display_*` methods for various levels of verbosity.

### CODE REVIEW

File: src/status/condition/mod.rs [hunk 1]
```diff
@@ -108,6 +108,58 @@ pub struct ClusterCondition {
...
+    /// Returns a short display string. This method wraps the
+    /// [`std::fmt::Display`] implementation of the [`ClusterCondition`].
+    pub fn display_short(&self) -> String {
+        self.to_string()
+    }
+
+    /// Returns a long display string. This method uses the
+    /// [`std::fmt::Display`] implementation of the [`ClusterCondition`] and
+    /// combines it with the optional message to provide more context.
+    pub fn display_long(&self) -> String {
```
Comment (sbernauer) [inline src/status/condition/mod.rs:155]: WDYT of a fn `display_long_only_on_err(&self) -> String`, that only appends the message when the condition is bad - which depends on condition?

Comment (Techassi) [reply src/status/condition/mod.rs:155]: Add the method `display_short_or_long` in bbaa7af. This either returns the long string when there is an error, or the short version when there is no error.

Comment (sbernauer) [reply src/status/condition/mod.rs:155]: I think we can not simply say "True is good, False is bard". We need to match the type I guess.

Comment (Techassi) [reply src/status/condition/mod.rs:155]: Yeah true. I feel like `ClusterConditions` are designed rather weird. I will come up with a solution which decides failure case based on the type and status.

Comment (Techassi) [reply src/status/condition/mod.rs:181]: I added an `is_error` method which determines if the condition is an error based on the type and status.

File: src/status/condition/mod.rs [hunk 2]
```diff
@@ -108,6 +108,98 @@ pub struct ClusterCondition {
...
+            },
+        };
+
+        out.fmt(f)
+    }
+}
+
+impl ClusterCondition {
+    /// Returns if the [`ClusterCondition`] is considered to be an error.
+    pub fn is_error(&self) -> bool {
```
Comment (sbernauer) [inline src/status/condition/mod.rs:147]: Like this method! Im just struggling to find a better word than "error" (as I would not consider startup a error). Maybe we want to rename it to is_good() and invert the logic

File: src/status/condition/mod.rs [hunk 3]
```diff
@@ -108,6 +108,98 @@ pub struct ClusterCondition {
...
+    /// [`std::fmt::Display`] implementation of the [`ClusterCondition`] and
+    /// combines it with the optional message to provide more context.
+    pub fn display_long(&self) -> String {
+        match &self.message {
+            Some(message) => format!("{}: {}", self, message),
+            None => self.to_string(),
+        }
+    }
+
+    /// Returns either a short ot long display string, This method additionally
```
Comment (sbernauer) [inline src/status/condition/mod.rs:188]: Please also update the docs to match the new solution

Comment (Techassi) [reply src/status/condition/mod.rs]: Updated doc comment in 9d3bfa9

File: src/status/condition/mod.rs [hunk 4]
```diff
@@ -108,6 +108,98 @@ pub struct ClusterCondition {
...
+        };
+
+        out.fmt(f)
+    }
+}
+
+impl ClusterCondition {
+    /// Returns if the [`ClusterCondition`] is considered to be in a good /
+    /// healthy state.
+    pub fn is_good(&self) -> bool {
```
Comment (sbernauer) [inline src/status/condition/mod.rs:148]: Just wanted to make sure it's explicitly made public. No need to comment ;)

**Review by sbernauer — APPROVED** [review]
Thx!

File: src/status/condition/mod.rs (hunk 4 above)
Comment (Techassi) [reply src/status/condition/mod.rs:148]: Yes. This allows outside code (for example `stackablectl`) to query this information. This is i.e. used for the colored output.

**Pull request closed without merging**